world/stat_checks/check_maker.py

```diff
@@ -25,10 +25,10 @@ def __init__(
         stat=None,
         skill=None,
         rating: DifficultyRating = None,
-        receivers: list = None,
+        receivers: list = [],
```

Comment:
So there's a real gotcha here in python for mutable default arguments. Because the signature of a function is in the class, it's in the global space and is allocated at import time. What this means is that there's a list as the default argument - and that list is shared by everyone who ever calls that function. So if it changes in any way, it'll affect other people who call the function.

For example, if someone were to have the default argument for receivers and then append something to receivers elsewhere, the next call that has the default argument would have the modified version. It can be a source of super confusing errors. tldr - use `None` like you did before and set the assigned class attribute to an empty list like you do in line 31

Reply:
I appreciate the sage advice.  I need to go find a book on Python that matches some of my C++ library regarding dos and don'ts.